extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/resteasy/OpenTelemetryServerFilter.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.opentelemetry.runtime.tracing.intrumentation.resteasy;
+
+import java.io.IOException;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.instrumentation.api.instrumenter.LocalRootSpan;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+/**
+ * Handles both RESTEasy Classic and RESTEasy Reactive (via Vert.x)
+ */
+@Provider
+public class OpenTelemetryServerFilter implements ContainerRequestFilter {
+
+    @Context
+    ResourceInfo resourceInfo;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        Span localRootSpan = LocalRootSpan.current();
+
+        localRootSpan.setAttribute(SemanticAttributes.CODE_NAMESPACE, resourceInfo.getResourceClass().getName());
+        localRootSpan.setAttribute(SemanticAttributes.CODE_FUNCTION, resourceInfo.getResourceMethod().getName());
```

Comment:
In RESTEasy Reactive, this is something we don't want to do as it leads to use of reflection (for no good reason).

In RESTEasy Reactive you can use `SimpleResourceInfo` instead of `ResourceInfo`

Reply:
@geoand I've spilt the Filters, one for RESTEasy Classic and one for RESTEasy Reactive.
can you please re-review?